opensrp-core/src/main/java/org/smartregister/account/AccountHelper.java

```diff
@@ -157,8 +117,61 @@ public static String getCachedOAuthToken(String accountName, String accountType,
     public static AccountManagerFuture<Bundle> reAuthenticateUserAfterSessionExpired(String accountName, String accountType, String authTokenType) {
         Account account = getOauthAccountByNameAndType(accountName, accountType);
         return accountManager.updateCredentials(account, authTokenType, null, null, null, null);
+    }
+
+    /**
+     * A Helper method to check if the Access Token is valid
+     */
+    public static boolean isAccessTokenValid(String accountName, String accountType) {
+        String createdAt = getAccountManagerValue(INTENT_KEY.ACCOUNT_ACCESS_TOKEN_CREATED_AT, accountName, accountType);
+        String accountExpires = getAccountManagerValue(INTENT_KEY.ACCOUNT_REFRESH_TOKEN_EXPIRES_IN, accountName, accountType);
+        Long createdAtLong = createdAt != null ? Long.parseLong(createdAt) : null;
+        Long accountExpiresLong = accountExpires != null ? Long.parseLong(accountExpires) : null;
```

Comment:
Ternary operators seem redundant in this two cases

Reply:
The conversion is for the _String_ values _createdAt_ and _accountExpires_ to their _Long_ versions using the _Long.parseLong_ method. If either of them is _null_ it would throw a **NumberFormatException**

Reply:
oh okay. My bad then